### PR TITLE
TC-62 Work Item list

### PR DESCRIPTION
**Pull request by LinetTorrico**
TC-62 Work Item list
1. Including work item list feature
2. Work Item List
workitemlist

### CODE REVIEW

File: src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java [hunk 1]
```diff
@@ -23,6 +24,15 @@
     @FindBy(id = "roleMenu")
     private WebElement roleMenu;
 
+    @FindBy(id = "workItemMenu")
+    private WebElement workItemMenu;
+
+
```
Comment (marcelogaray) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:34]: please user format code
remove blank line

File: src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java [hunk 2]
```diff
@@ -36,4 +46,16 @@ public RoleList clickOnRoleMenu() {
...
+    public void clickOnWorkItemsMenu(){
+        CommonEvents.clickButton(workItemMenu);
+    }
+
+    public WorkItemList clickOnWorkItemMenu(){
+        CommonEvents.clickButton(workItemSubMenu);
+        return new WorkItemList();
+    }
+
+
```
Comment (marcelogaray) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java]: remove multiples black lines

File: src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java [hunk 3]
```diff
@@ -23,6 +24,15 @@
...
+    private WebElement workItemMenu;
+
+
+    @FindBy(id = "workItemSubMenu")
+    private WebElement workItemSubMenu;
+
+    //workItemSubMenu
+
     public void clickOnPersonnelMenu(){
         CommonEvents.clickButton(personnelMenu);
```
Comment (marcelogaray) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java]: it should be returned a page object according to purpose

File: src/main/java/org/umssdiplo/automationv01/core/managepage/workItem/WorkItemList.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ * @author Miguel Rojas
+ */
+package org.umssdiplo.automationv01.core.managepage.workItem;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.umssdiplo.automationv01.core.managepage.BasePage;
+import org.umssdiplo.automationv01.core.utils.CommonEvents;
+
+public class WorkItemList extends BasePage {
```
Comment (marcelogaray) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/workItem/WorkItemList.java:11]: this class needs a constructure

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 5]
```diff
@@ -25,23 +26,23 @@ public void ssiAHomePageIsLoaded() throws Throwable {
...
-    public void clickNewEmployeeBtn() throws Throwable{
+    public void clickNewEmployeeBtn() throws Throwable {
         employeeList.clickNewEmployeeButton();
     }
 
     @Then("'Employee List' page loads correctly")
-    public void isEmployeeListPresent() throws Throwable{
+    public void isEmployeeListPresent() throws Throwable {
         boolean result = employeeList.isEmployeeListPresent();
         Assert.assertTrue(result);
```
Comment (marcelogaray) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: when this assertion will fail then you will need to set an error message

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 6]
```diff
@@ -60,6 +61,27 @@ public void roleListIsShowedInPage() throws Throwable {
     }
 
     // Work Item List
+    private WorkItemList workItemList;
+
+    @And("Clicking on Work Items menu on 'Header' page")
+    public void clickWorkItemsMenu() throws Throwable {
+        ssiaHome.clickOnWorkItemsMenu();
```
Comment (marcelogaray) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: review your structure according to before observation

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 7]
```diff
@@ -60,6 +61,27 @@ public void roleListIsShowedInPage() throws Throwable {
...
+    }
+
+    @And("Clicking on NewWorkItem button")
+    public void clickNewWorkItemBtn() throws Throwable {
+        workItemList.clickNewWorkItemButton();
+    }
+
+    @Then("^'Work Item List' page loads correctly$")
+    public void workItemListIsShowedInPage() throws Throwable {
+        Assert.assertTrue(workItemList.isWorkItemListPresent(), "Fail, Work Item list is not loaded");
```
Comment (marcelogaray) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: workitmelist should be reviewed it.

**Review by marcelogaray — CHANGES_REQUESTED** [review]
review observations

**Comment by LinetTorrico** [comment]
1. please user format code remove blank line (Fixed)
2. it should be returned a page object according to purpose (Fixed)
3. this class needs a constructure (Fixed)
4. when this assertion will fail then you will need to set an error message (Fixed)
workitemlist

**Comment by LinetTorrico** [comment]
On and Clicking has been removed

File: src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java [hunk 8]
```diff
@@ -23,11 +25,19 @@
...
+    @FindBy(id = "workItemMenu")
+    private WebElement workItemMenu;
+
+
+    @FindBy(id = "workItemSubMenu")
+    private WebElement workItemSubMenu;
+
+
+    public void clickOnPersonnelMenu() {
         CommonEvents.clickButton(personnelMenu);
```
Comment (marcelogaray) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java]: please rename this method clickPersonnelMenu

File: src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java [hunk 9]
```diff
@@ -36,4 +46,14 @@ public RoleList clickOnRoleMenu() {
         CommonEvents.clickButton(roleMenu);
         return new RoleList();
     }
+
+    public void clickOnWorkItemsMenu() {
```
Comment (marcelogaray) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java]: remove this method, it is not return an object.

File: src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java [hunk 10]
```diff
@@ -36,4 +46,14 @@ public RoleList clickOnRoleMenu() {
...
+        CommonEvents.clickButton(workItemMenu);
+    }
+
+    public WorkItemList clickOnWorkItemMenu() {
```
Comment (marcelogaray) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java]: please rename this method clickWorkItemMenu

**Review by marcelogaray — CHANGES_REQUESTED** [review]
review observations

**Comment by LinetTorrico** [comment]
Corrections has been fixed 
corrections 06-22-2018

File: src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java (hunk 1 above)
Comment (marcelogaray) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:34]: I see multiple blank lines, please fix it.

File: src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java [hunk 11]
```diff
@@ -44,6 +52,12 @@ public RoleList clickOnRoleMenu() {
     public SafetyMenu clickSafetyMenu() {
         CommonEvents.clickButton(safetyMenu);
         return new SafetyMenu();
+
```
Comment (marcelogaray) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:55]: remove blank line

File: src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java [hunk 12]
```diff
@@ -44,6 +52,12 @@ public RoleList clickOnRoleMenu() {
...
     }
 
+    public WorkItemsMenu clickWorkItemsMenu() {
+        CommonEvents.clickButton(workItemMenu);
+        return new WorkItemsMenu();
+
```
Comment (marcelogaray) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:61]: remove blank line

File: src/main/java/org/umssdiplo/automationv01/core/managepage/menuheader/workItemsMenu/WorkItemsMenu.java [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+public class WorkItemsMenu extends BasePage {
+
+    @FindBy(id = "workItemMenu")
+    private WebElement workItemMenu;
+
+    public WorkItemList clickAccidentMenu() {
+        CommonEvents.clickButton(workItemMenu);
+        return new WorkItemList();
+    }
+
```
Comment (marcelogaray) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/menuheader/workItemsMenu/WorkItemsMenu.java:18]: remove blank line

**Review by marcelogaray — CHANGES_REQUESTED** [review]
review observations

**Comment by marcelogaray** [comment]
Please fix conflicts and my observations
Send me a pull request once you fix it, I will close it in the meantime

**Pull request closed without merging**